### PR TITLE
doc: complete think in react

### PR DESCRIPTION
**Pull request by chloewlin**
Hi, this is my work on `thinking in react`. A few words I am not sure of: 

- affordance
- modular 模塊化/模組化
- implement 實作
- react doc 文件
- representation 
- checkbox 複選框 
- attribute / property 
- callbacks

### CODE REVIEW

**Comment by bugtender** [comment]
Hi, @chloewlin Thanks!

we defined
* implement 實作 -> #16 
* attribute 屬性 -> #13 

and I think..

* affordance -> 直觀功能？ [ref](https://zh.wikipedia.org/wiki/%E6%89%BF%E6%93%94%E7%89%B9%E8%B3%AA)
* modular -> 模組化
* react doc -> react 文件
* representation -> 呈現方式？

and should keep.
* keep checkbox -> it's an input type
* keep property -> we keep properties
* keep callbacks 

last, maybe `數據` would be `資料`?

File: content/docs/thinking-in-react.md [hunk 1]
```diff
@@ -31,118 +31,119 @@ Our JSON API returns some data that looks like this:
...
+如果你試著在範例目前的版本中印出或勾選複選框，你會看到 React 無視你的輸入。這是刻意的，因為我們把 `input` 的 `value` prop 設定為永遠和從 `FilterableProductTable` 傳下來的 `state` ㄧ樣。
 
-Let's think about what we want to happen. We want to make sure that whenever the user changes the form, we update the state to reflect the user input. Since components should only update their own state, `FilterableProductTable` will pass callbacks to `SearchBar` that will fire whenever the state should be updated. We can use the `onChange` event on the inputs to be notified of it. The callbacks passed by `FilterableProductTable` will call `setState()`, and the app will be updated.
+讓我們思考一下我們想要做些什麼。我們想確保當使用者改變這個表格時，我們會更新 state 以反映使用者的輸入。既然 component 只應該更新它自己本身的 state， `FilterableProductTable` 將會把 callback 傳給 `SearchBar`，而它們則會在 state 該被更新的時候被觸發。我們可以在輸入上使用 `onChange` 這個 event 來
+接收通知。被 `FilterableProductTable` 傳下來的 callback 則會呼叫 `setState()`，之後應用程式就會被更新。
 
-Though this sounds complex, it's really just a few lines of code. And it's really explicit how your data is flowing throughout the app.
+雖然這聽起來負責，但實際上這所用的程式碼很少。而你的數據在應用程式中流動的方向是非常明確的。
```
Comment (zeroshine) [inline content/docs/thinking-in-react.md]: 複雜

**Comment by chloewlin** [comment]
@bugtender Thank you for your feedback! Can you take one final quick look on my revisions?

File: content/docs/thinking-in-react.md (hunk 1 above)
Comment (chloewlin) [reply content/docs/thinking-in-react.md]: Thank you @zeroshine

**Comment by chloewlin** [comment]
@bugtender Thank you! The conflicts are resolved.

**Comment by chloewlin** [comment]
Thank you @bugtender! I am going to merge this in now 🥇

**Pull request merged**